app/models/episode.rb

```diff
@@ -19,11 +26,20 @@ def update(args)
 
   class << self
     def prepare_params(args)
-      raise(ActiveRecord::RecordInvalid.new, 'Invalid video') unless args[:video].match VALID_YOUTUBE_LINK
+      if args[:video] && args[:date]
+        raise(ActiveRecord::RecordInvalid.new, 'Invalid video') unless args[:video].match VALID_YOUTUBE_LINK
 
-      args[:video] = args[:video].match(YOUTUBE_VIDEO_IDENTIFIER)[5]
-      args[:date] = Time.parse(args[:date]).utc
+        args[:video] = args[:video].match(YOUTUBE_VIDEO_IDENTIFIER)[5]
+        args[:date] = Time.parse(args[:date]).utc
+      end
       args
     end
   end
+
+  private
+
+  def check_youtube_status(video)
+    @video = Yt::Video.new id: video
```

Comment:
same

Reply:
This method is called in the update_youtube_status, so I just added video check there